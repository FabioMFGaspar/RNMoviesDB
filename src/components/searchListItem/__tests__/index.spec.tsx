import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {mockMovies} from '../../../mocks/movieDB.mock';
import SearchListItem from '..';

describe('SearchListItem test suite', () => {
  const mockHandle = jest.fn();
  const defaultProps = {
    movie: mockMovies[0],
    handlePress: mockHandle,
  };

  it('Default snapshot test', () => {
    const {toJSON} = render(<SearchListItem {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should find movie title', () => {
    const {queryByText} = render(<SearchListItem {...defaultProps} />);

    const movie = queryByText('The Avengers');

    expect(movie).toBeTruthy();
  });

  it('Should trigger onPress', () => {
    const {getByText} = render(<SearchListItem {...defaultProps} />);

    const movie = getByText('The Avengers');
    fireEvent.press(movie);

    expect(mockHandle).toHaveBeenCalledWith(defaultProps.movie.id);
  });
});
