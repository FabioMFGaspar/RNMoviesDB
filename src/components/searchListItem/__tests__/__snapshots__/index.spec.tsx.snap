// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchListItem test suite Default snapshot test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2b2b3b",
      "borderRadius": 10,
      "display": "flex",
      "marginHorizontal": 15,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w400/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 200,
        "width": 360,
      }
    }
  />
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#e6e6ff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      The Avengers
    </Text>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "#8c8c8c",
          "fontSize": 12,
        }
      }
    >
      When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!
    </Text>
  </View>
</View>
`;
